test/integration/cookbooks/resolver-test/recipes/default.rb

```diff
@@ -1,4 +1,9 @@
-node.default['resolver']['search'] = ['localdomain']
-node.default['resolver']['domain'] = nil
-node.default['resolver']['nameservers'] = %w(1.1.1.1 8.8.8.8)
-node.default['resolver']['options'] = { 'timeout' => 2, 'rotate' => nil }
+resolver_config '/etc/resolv.conf' do
+  nameservers ['1.1.1.1', '1.0.0.1']
+  domain 'test.com'
+  search ['test1.com', 'test2.com']
+  options(
+    'timeout' => 2
+  )
+  override_system_configuration true
+end
```

Comment:
For tests, could we install dig? On ubuntu it's part of the dnsutils package, fedora it's bind-utils, freebsd it's dns/bind-tools. It may be pre-installed already on some of these though.

Reply:
That was my initial idea but the differentiation in package names put me off on the cost/benefit ratio.

Reply:
I think the benefit here is pretty high though. curl just tests that dns resolution succeeds, not that it's hitting the desired nameserver. It's fairly straightforward to handle different names with a case statement on ohai os info.